homeassistant/components/bsblan/config_flow.py

```diff
@@ -0,0 +1,137 @@
+            data={
+                CONF_HOST: user_input[CONF_HOST],
+                CONF_PORT: user_input[CONF_PORT],
+                CONF_DEVICE_IDENT: info.device_identification,
+            },
+        )
+
+    async def async_step_zeroconf(
+        self, user_input: Optional[ConfigType] = None
+    ) -> Dict[str, Any]:
+        """Handle zeroconf discovery."""
+        if user_input is None:
+            return self.async_abort(reason="connection_error")
+
+        # Hostname is format: my-ke.local.
+        host = user_input["hostname"].rstrip(".")
+        try:
+            info = await self._get_bsblan_info(host, user_input[CONF_PORT])
+        except BSBLanError:
+            return self.async_abort(reason="connection_error")
+
+        # Check if already configured
+        await self.async_set_unique_id(info.device_identification)
+        self._abort_if_unique_id_configured()
+
+        # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+        self.context.update(
+            {
+                CONF_HOST: host,
+                CONF_PORT: user_input[CONF_PORT],
+                CONF_DEVICE_IDENT: info.device_identification,
+                "title_placeholders": {"device_ident": info.device_identification},
+            }
+        )
+
+        # Prepare configuration flow
+        return self._show_confirm_dialog()
+
+    # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+    async def async_step_zeroconf_confirm(
+        self, user_input: ConfigType = None
+    ) -> Dict[str, Any]:
+        """Handle a flow initiated by zeroconf."""
+        if user_input is None:
+            return self._show_confirm_dialog()
+
+        try:
+            info = await self._get_bsblan_info(
+                self.context.get(CONF_HOST), self.context.get(CONF_PORT)
+            )
+        except BSBLanError:
+            return self.async_abort(reason="connection_error")
+
+        # Check if already configured
+        await self.async_set_unique_id(info.device_identification)
+        self._abort_if_unique_id_configured()
+
+        return self.async_create_entry(
+            title=self.context.get(CONF_DEVICE_IDENT),
+            data={
+                CONF_HOST: self.context.get(CONF_HOST),
+                CONF_PORT: self.context.get(CONF_PORT),
+                CONF_DEVICE_IDENT: self.context.get(CONF_DEVICE_IDENT),
+            },
+        )
+
+    def _show_setup_form(self, errors: Optional[Dict] = None) -> Dict[str, Any]:
+        """Show the setup form to the user."""
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST): str,
+                    vol.Optional(CONF_PORT, default=80): int,
+                }
+            ),
+            errors=errors or {},
+        )
+
+    def _show_confirm_dialog(self, errors: Optional[Dict] = None) -> Dict[str, Any]:
```

Comment:
We can inline the body in `async_step_zeroconf_confirm` instead.

Reply:
I am not sure if all is ok now. Let me know please